.github/workflows/arm64-build-checks.yml

```diff
@@ -0,0 +1,57 @@
+name: arm64 build checks
+
+on:
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-18.04
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install dependency package
+      run: |
+        sudo sed -i -E 's|^deb ([^ ]+) (.*)$|deb [arch=amd64] \1 \2\ndeb [arch=arm64] http://ports.ubuntu.com/ubuntu-ports/ \2|' /etc/apt/sources.list
+        sudo dpkg --add-architecture arm64
+        sudo apt-get update
+        sudo apt-get install -y --no-install-recommends \
+                crossbuild-essential-arm64 \
+                git \
+                cmake \
+                ca-certificates \
+                libpython-dev:arm64 \
+                libpython3-dev:arm64 \
+                libusb-1.0-0-dev:arm64 \
+                python-numpy \
+                python3-numpy \
+                libgstreamer1.0-dev:arm64 \
+                libgstreamer-plugins-base1.0-dev:arm64
+
+    - name: Fetch opencv_contrib
+      run: |
+        cd ..
+        git clone --depth 1 https://github.com/opencv/opencv_contrib.git
+
+    - name: Configure
+      run: |
+        mkdir build
+        cd build
+        cmake -DBUILD_SHARED_LIBS=ON \
+              -DBUILD_PERF_TESTS=ON \
+              -DBUILD_TESTS=ON \
+              -DWITH_OPENNI2=ON \
+              -DWITH_GDAL=ON \
+              -DBUILD_EXAMPLES=OFF \
+              -DPYTHON_DEFAULT_EXECUTABLE=/usr/bin/python3 \
+              -DWITH_OPENCL=OFF \
+              -DBUILD_DOCS=OFF \
+              -DCMAKE_TOOLCHAIN_FILE=../platforms/linux/aarch64-gnu.toolchain.cmake \
+              -DWITH_CUDA=OFF \
```

Comment:
Please check if `WITH_CUDA`, `WITH_OPENCL` and other flags already have these values by default

Reply:
Actually, I got these flags from [existing arm64 build](https://pullrequest.opencv.org/buildbot/builders/master-contrib_ARMv8-lin), so I just keep it same with [link](https://pullrequest.opencv.org/buildbot/builders/master-contrib_ARMv8-lin/builds/11233/steps/cmake/logs/stdio).

But I think you are right, I should cleanup some redundant flag:

- [BUILD_SHARED_LIBS](https://github.com/opencv/opencv/blob/master/CMakeLists.txt#L438): NOT (ANDROID OR APPLE_FRAMEWORK), it would be ON in ARMv8, **will remove it**.
- [BUILD_PERF_TESTS](https://github.com/opencv/opencv/blob/master/CMakeLists.txt#L446): NOT INSTALL_CREATE_DISTRIB  IF (NOT APPLE_FRAMEWORK) ), it would be ON in ARMv8, **will remove it**.
- [BUILD_TESTS](https://github.com/opencv/opencv/blob/master/CMakeLists.txt#L447): NOT INSTALL_CREATE_DISTRIB  IF (NOT APPLE_FRAMEWORK) ), it would be ON in ARMv8, **will remove it**.
- [WITH_OPENNI2](https://github.com/opencv/opencv/blob/master/CMakeLists.txt#L312): OFF by default, so it's okay to keep it ON.
- [WITH_GDAL](https://github.com/opencv/opencv/blob/master/CMakeLists.txt#L399): OFF by default, so it's okay to keep it ON.
- [BUILD_EXAMPLES](https://github.com/opencv/opencv/blob/master/CMakeLists.txt#L444): OFF by default, **will remove it**.
- [WITH_OPENCL](https://github.com/opencv/opencv/blob/master/CMakeLists.txt#L369): (NOT ANDROID AND NOT CV_DISABLE_OPTIMIZATION), ON in ARMv8, so it's okay to keep it OFF.
- [BUILD_DOCS](https://github.com/opencv/opencv/blob/master/CMakeLists.txt#L443): OFF  IF (NOT WINRT AND NOT APPLE_FRAMEWORK)), OFF in ARMv8, **will remove it**.
- [WITH_CUDA](https://github.com/opencv/opencv/blob/master/CMakeLists.txt#L243): OFF in ARMv8, **will remove it**.

So, the finally flag list looks like:
```
cmake -DWITH_OPENNI2=ON \
      -DWITH_GDAL=ON \
      -DPYTHON_DEFAULT_EXECUTABLE=/usr/bin/python3 \
      -DWITH_OPENCL=OFF \
      -DCMAKE_TOOLCHAIN_FILE=../platforms/linux/aarch64-gnu.toolchain.cmake \
      -DOPENCV_EXTRA_MODULES_PATH=../../opencv_contrib/modules ../
```

Reply:
I believe that `WITH_OPENNI2` , `WITH_GDAL` and `WITH_OPENCL` are also extra

Reply:
done